src/constructs/VPC.ts

```diff
@@ -0,0 +1,50 @@
+import { SecurityGroup, Vpc } from "@aws-cdk/aws-ec2";
+import { Construct as CdkConstruct } from "@aws-cdk/core";
+import { FromSchema } from "json-schema-to-ts";
+import { AwsCfInstruction } from "@serverless/typescript";
+import { AwsConstruct, AwsProvider } from "../classes";
+
+const VPC_DEFINITION = {
+    type: "object",
+    properties: {
+        type: { const: "vpc" },
+    },
+    additionalProperties: false,
+    required: [],
+} as const;
+
+type Configuration = FromSchema<typeof VPC_DEFINITION>;
+
+export class VPC extends AwsConstruct {
+    public static type = "vpc";
+    public static schema = VPC_DEFINITION;
+
+    private readonly vpc: Vpc;
+
+    constructor(scope: CdkConstruct, id: string, configuration: Configuration, private provider: AwsProvider) {
+        super(scope, id);
+
+        this.vpc = new Vpc(this, "VPC", {
+            maxAzs: 2,
+        });
+
+        const privateSubnets = this.vpc.privateSubnets;
```

Comment:
Also maybe we would like to create isolated subnets to put services into (like RDS, ElastiCache, etc), instead of putting them in the private subnets.

Reply:
Let's put that aside for now, putting these services in the private subnets should be a good fit for most use cases.

@deleugpn mentioned that isolated subnets could be useful for security compliance, but these are advanced use cases.